from pyqtgraph import GraphicsLayoutWidget
from PyQt4 import QtCore, QtGui

"""
This file is GUI-Design. If you install Qt-Designer and open 'GUI_Design.ui' file, it would be much convenient to modify them.
"""


try:
    _fromUtf8 = QtCore.QString.fromUtf8
except AttributeError:
    def _fromUtf8(s):
        return s

try:
    _encoding = QtGui.QApplication.UnicodeUTF8
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig, _encoding)
except AttributeError:
    def _translate(context, text, disambig):
        return QtGui.QApplication.translate(context, text, disambig)

class Ui_Dialog(object):
    def setupUi(self, Dialog):
        Dialog.setObjectName(_fromUtf8("Dialog"))
        Dialog.resize(800, 580)
        icon = QtGui.QIcon()
        icon.addPixmap(QtGui.QPixmap(_fromUtf8("C:/Users/fieldcontrol/Degausser_final/icon.png")), QtGui.QIcon.Normal, QtGui.QIcon.Off)
        Dialog.setWindowIcon(icon)
        self.group_Message = QtGui.QGroupBox(Dialog)
        self.group_Message.setGeometry(QtCore.QRect(370, 130, 411, 141))
        self.group_Message.setObjectName(_fromUtf8("group_Message"))
        self.MessageBox = QtGui.QTextBrowser(self.group_Message)
        self.MessageBox.setGeometry(QtCore.QRect(10, 20, 391, 111))
        self.MessageBox.setObjectName(_fromUtf8("MessageBox"))
        self.group_B = QtGui.QGroupBox(Dialog)
        self.group_B.setGeometry(QtCore.QRect(0, 270, 781, 301))
        self.group_B.setObjectName(_fromUtf8("group_B"))
        self.StopInput = QtGui.QPushButton(self.group_B)
        self.StopInput.setGeometry(QtCore.QRect(680, 265, 91, 31))
        self.StopInput.setObjectName(_fromUtf8("StopInput"))
        self.Measure = QtGui.QPushButton(self.group_B)
        self.Measure.setGeometry(QtCore.QRect(580, 265, 91, 31))
        self.Measure.setObjectName(_fromUtf8("Measure"))
        self.checkX = QtGui.QCheckBox(self.group_B)
        self.checkX.setGeometry(QtCore.QRect(10, 270, 121, 21))
        self.checkX.setText(_fromUtf8("X-axis"))
        self.checkX.setChecked(True)
        self.checkX.setObjectName(_fromUtf8("checkX"))
        self.checkX.setStyleSheet("font-size:15pt; color: red")
        self.checkZ = QtGui.QCheckBox(self.group_B)
        self.checkZ.setGeometry(QtCore.QRect(190, 270, 71, 19))
        self.checkZ.setChecked(True)
        self.checkZ.setObjectName(_fromUtf8("checkZ"))
        self.checkZ.setStyleSheet("font-size:15pt; color: blue")
        self.checkY = QtGui.QCheckBox(self.group_B)
        self.checkY.setGeometry(QtCore.QRect(100, 270, 71, 19))
        self.checkY.setChecked(True)
        self.checkY.setObjectName(_fromUtf8("checkY"))
        self.checkY.setStyleSheet("font-size:15pt; color: yellow")
        self.MeasuredB = GraphicsLayoutWidget(self.group_B)
        self.MeasuredB.setGeometry(QtCore.QRect(10, 20, 761, 241))
        self.MeasuredB.setObjectName(_fromUtf8("MeasuredB"))
        self.checkflux = QtGui.QCheckBox(self.group_B)
        self.checkflux.setGeometry(QtCore.QRect(280, 270, 101, 20))
        self.checkflux.setLayoutDirection(QtCore.Qt.RightToLeft)
        self.checkflux.setChecked(True)
        self.checkflux.setObjectName(_fromUtf8("checkflux"))
        self.group_Wave = QtGui.QGroupBox(Dialog)
        self.group_Wave.setGeometry(QtCore.QRect(0, 10, 371, 261))
        self.group_Wave.setObjectName(_fromUtf8("group_Wave"))
        self.group_Settings = QtGui.QGroupBox(Dialog)
        self.group_Settings.setGeometry(QtCore.QRect(370, 10, 411, 121))
        self.group_Settings.setObjectName(_fromUtf8("group_Settings"))
        self.Amp_l = QtGui.QLabel(self.group_Settings)
        self.Amp_l.setGeometry(QtCore.QRect(10, 20, 71, 21))
        self.Amp_l.setLayoutDirection(QtCore.Qt.LeftToRight)
        self.Amp_l.setObjectName(_fromUtf8("Amp_l"))
        self.Amp = QtGui.QLineEdit(self.group_Settings)
        self.Amp.setGeometry(QtCore.QRect(90, 20, 31, 21))
        self.Amp.setObjectName(_fromUtf8("Amp"))
        self.Freq_l = QtGui.QLabel(self.group_Settings)
        self.Freq_l.setGeometry(QtCore.QRect(160, 20, 71, 21))
        self.Freq_l.setObjectName(_fromUtf8("Freq_l"))
        self.Freq = QtGui.QLineEdit(self.group_Settings)
        self.Freq.setGeometry(QtCore.QRect(240, 20, 31, 21))
        self.Freq.setObjectName(_fromUtf8("Freq"))
        self.Dur_l = QtGui.QLabel(self.group_Settings)
        self.Dur_l.setGeometry(QtCore.QRect(310, 20, 61, 21))
        self.Dur_l.setObjectName(_fromUtf8("Dur_l"))
        self.Dur = QtGui.QLineEdit(self.group_Settings)
        self.Dur.setGeometry(QtCore.QRect(370, 20, 31, 21))
        self.Dur.setObjectName(_fromUtf8("Dur"))
        self.Keep_l = QtGui.QLabel(self.group_Settings)
        self.Keep_l.setGeometry(QtCore.QRect(10, 40, 71, 21))
        self.Keep_l.setObjectName(_fromUtf8("Keep_l"))
        self.Keep = QtGui.QLineEdit(self.group_Settings)
        self.Keep.setGeometry(QtCore.QRect(90, 40, 31, 21))
        self.Keep.setObjectName(_fromUtf8("Keep"))
        self.Offset_l = QtGui.QLabel(self.group_Settings)
        self.Offset_l.setGeometry(QtCore.QRect(160, 40, 51, 21))
        self.Offset_l.setObjectName(_fromUtf8("Offset_l"))
        self.Offset = QtGui.QLineEdit(self.group_Settings)
        self.Offset.setGeometry(QtCore.QRect(240, 40, 31, 21))
        self.Offset.setObjectName(_fromUtf8("Offset"))
        self.Dev_l = QtGui.QLabel(self.group_Settings)
        self.Dev_l.setGeometry(QtCore.QRect(10, 60, 51, 21))
        self.Dev_l.setObjectName(_fromUtf8("Dev_l"))
        self.Dev = QtGui.QLineEdit(self.group_Settings)
        self.Dev.setGeometry(QtCore.QRect(90, 60, 31, 21))
        self.Dev.setObjectName(_fromUtf8("Dev"))
        self.Port_l = QtGui.QLabel(self.group_Settings)
        self.Port_l.setGeometry(QtCore.QRect(160, 60, 31, 21))
        self.Port_l.setObjectName(_fromUtf8("Port_l"))
        self.Port = QtGui.QLineEdit(self.group_Settings)
        self.Port.setGeometry(QtCore.QRect(240, 60, 31, 21))
        self.Port.setObjectName(_fromUtf8("Port"))
        self.FormUpdate = QtGui.QPushButton(self.group_Settings)
        self.FormUpdate.setGeometry(QtCore.QRect(310, 50, 91, 31))
        self.FormUpdate.setObjectName(_fromUtf8("FormUpdate"))
        self.progressBar = QtGui.QProgressBar(self.group_Settings)
        self.progressBar.setGeometry(QtCore.QRect(10, 90, 301, 23))
        self.progressBar.setProperty("value", 0)
        self.progressBar.setObjectName(_fromUtf8("progressBar"))
        self.StartButton = QtGui.QPushButton(self.group_Settings)
        self.StartButton.setGeometry(QtCore.QRect(310, 85, 91, 31))
        self.StartButton.setObjectName(_fromUtf8("StartButton"))
        self.waveform = GraphicsLayoutWidget(Dialog)
        self.waveform.setGeometry(QtCore.QRect(10, 30, 351, 231))
        self.waveform.setObjectName(_fromUtf8("waveform"))

        self.retranslateUi(Dialog)
        QtCore.QMetaObject.connectSlotsByName(Dialog)


    def retranslateUi(self, Dialog):
        Dialog.setWindowTitle(_translate("Dialog", "Degaussing and Magnetic Field Generator", None))
        self.group_Message.setTitle(_translate("Dialog", "Message Box", None))
        self.group_B.setTitle(_translate("Dialog", "Measured Magnetic Field", None))
        self.StopInput.setText(_translate("Dialog", "Stop Input", None))
        self.checkZ.setText(_translate("Dialog", "Z-axis", None))
        self.checkY.setText(_translate("Dialog", "Y-axis", None))
        self.checkX.setText(_translate("Dialog", "X-axis", None))
        self.Measure.setText(_translate("Dialog", "Measure", None))
        self.checkflux.setText(_translate("Dialog", "Fluxgate", None))
        self.group_Wave.setTitle(_translate("Dialog", "Expected Wave Form", None))
        self.group_Settings.setTitle(_translate("Dialog", "Settings", None))
        self.Amp_l.setText(_translate("Dialog", "Amplitude", None))
        self.Amp.setText(_translate("Dialog", "1", None))
        self.Freq_l.setText(_translate("Dialog", "Frequency", None))
        self.Freq.setText(_translate("Dialog", "10", None))
        self.Dur_l.setText(_translate("Dialog", "Duration", None))
        self.Dur.setText(_translate("Dialog", "20", None))
        self.Keep_l.setText(_translate("Dialog", "Keep Time", None))
        self.Keep.setText(_translate("Dialog", "5", None))
        self.Offset_l.setText(_translate("Dialog", "Offset", None))
        self.Offset.setText(_translate("Dialog", "0", None))
        self.Dev_l.setText(_translate("Dialog", "Device", None))
        self.Dev.setText(_translate("Dialog", "1", None))
        self.Port_l.setText(_translate("Dialog", "Port", None))
        self.Port.setText(_translate("Dialog", "1", None))
        self.FormUpdate.setText(_translate("Dialog", "Update ", None))
        self.StartButton.setText(_translate("Dialog", '&Generate', None))


